prereise/gather/data/remap_ba_area/remap_ba_area.py

```diff
@@ -0,0 +1,270 @@
+import json
+import os
+import shutil
+
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import pandas as pd
+import requests
+
+from prereise.gather.data.remap_ba_area.map_data import (
+    miso_name_map,
+    spp_name_map,
+    states_to_map_with_counties,
+)
+
+
+def get_full_filename(file_name):
+    return os.path.join(os.path.dirname(__file__), file_name)
+
+
+def compose_ba_map(output_shape_file="remap_ba_area.shp"):
+    """
+    Create the control area and subregion map. For CAs heavily overlapped, we define several zones and match the CAs to
+    these zones. The ca_source name is either mapped to one zone in BA_map_name, or allocated to multiple zones in
+    BA_map_name with fractions based on the area.
+    The function generates the remap_ba_area shape file and remap_ba_area.tif file.
+
+    :param str output_shape_file: output shape file name
+    :return: (*geopandas.GeoDataFrame*)  -- the final control zone map
+    """
+
+    with open(get_full_filename("../BA_County_map.json")) as json_file:
+        ba2county_dict = json.load(json_file)
+    # Add two missing counties
+    # this may be fixed in future updates of BA_County_map.json
+
+    # Turn dict into pandas table
+    county2ba_dict = {
+        county: ba
+        for (ba, ba_counties) in ba2county_dict.items()
+        for county in ba_counties
+    }
+    county2ba = pd.Series(county2ba_dict).to_frame().reset_index()
+    county2ba.columns = ["index", "BA"]
+    county2ba[["county", "state"]] = county2ba["index"].str.split(
+        "__", n=1, expand=True
+    )
+    county2ba = county2ba.drop(columns=["index"])
+    county2ba.set_index(["county", "state"], inplace=True)
+
+    # US Counties Shapefile
+    counties_shp_file = gpd.read_file(get_full_filename("cb_2020_us_county_500k.zip"))
+    # Remove AK, HI, PR, etc.
+    counties_shp_file = counties_shp_file[["NAME", "STUSPS", "geometry"]].rename(
+        columns={"NAME": "county", "STUSPS": "state"}
+    )
+    lower48 = counties_shp_file.loc[
+        counties_shp_file["state"].isin(states_to_map_with_counties)
+    ]
+    lower48.set_index(["county", "state"], inplace=True)
+    counties_with_ba = lower48.join(county2ba)
+    # Since these all seem to be in the pandhandle, map them to ERCOT north
+    counties_with_ba.loc[counties_with_ba["BA"] == "ERCOT", "BA"] = "ERCO-N"
+    # Aggregate counties into BA
+    ba_dissolved = counties_with_ba.dissolve(by="BA")["geometry"].to_frame()
+    # Drop PJM, replace with alternate shapes
+    ba_dissolved.drop("PJM", inplace=True)
+    pjm = gpd.read_file(get_full_filename("pjm.zip")).set_index("BA")
+    ba_dissolved = pd.concat([ba_dissolved, pjm])
+    california = gpd.read_file(
+        os.path.join(
+            os.path.dirname(__file__),
+            "CA_balancing_authorities.zip",
+        )
+    )
```

Comment:
Is this equivalent to `california = gpd.read_file("CA_balancing_authorities.zip")`?

Reply:
The python -m .. run will have an issue to find the zip file. To use `__file_` is a safer way if we use github to distribute the zip file.

Reply:
Maybe I'm missing something here. It seems like calling `get_full_filename("CA_balancing_authorities.zip")` will return `os.path.join(os.path.dirname(__file__), "CA_balancing_authorities.zip")`, which is exactly what we have explicitly in the code here. It seems like we also use `get_full_filename` on other ZIP files are well, e.g. `get_full_filename("cb_2020_us_county_500k.zip")`. Is there something special about this zip file that differs from the others?